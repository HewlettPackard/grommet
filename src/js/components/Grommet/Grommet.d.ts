import * as React from 'react';
import Grommet from '../index';

export declare namespace GrommetTypes {
  type Direction = 'rtl' | 'ltr';
}

export interface GrommetProps extends Grommet.Props {
  dir: GrommetTypes.Direction;
  theme: {};
}

export class Grommet extends React.Component<GrommetProps, undefined> { }

export default Grommet;
