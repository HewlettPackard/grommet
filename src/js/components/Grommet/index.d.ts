import * as React from "react";

export interface GrommetProps {
  full?: boolean;
  theme?: object;
}

declare const Grommet: React.StatelessComponent<GrommetProps>;

export { Grommet };
