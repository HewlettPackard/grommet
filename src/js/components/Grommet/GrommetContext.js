import React from 'react';

// Track which components mounted onto the DOM are coming from
// Grommet (see FocusedContainer.js)
export const GrommetContext = React.createContext(undefined);
