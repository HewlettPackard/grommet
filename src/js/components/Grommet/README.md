## Grommet
This is the top level Grommet container.

[![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-site?initialpath=grommet&amp;module=%2Fscreens%2FGrommet.js)
## Usage

```javascript
import { Grommet } from 'grommet';
<Grommet>...</Grommet>
```

## Properties

**dir**

Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.

```
rtl
ltr
```

**theme**

Custom styles for Grommet app component.

```
object
```
  