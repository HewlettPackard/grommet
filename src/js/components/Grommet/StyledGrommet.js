import styled, { css } from 'styled-components';

import { baseStyle } from '../../utils';

const fullStyle = css`
  width: 100vw;
  height: 100vh;
`;

const StyledGrommet = styled.div`
  ${baseStyle}
  ${props => props.full && fullStyle}
  ${props => props.theme.global.font.face}
`;

export default StyledGrommet.extend`
  ${props => props.theme.grommet.extend}
`;
