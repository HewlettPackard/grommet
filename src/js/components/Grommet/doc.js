import { describe, PropTypes } from 'react-desc';

import { getAvailableAtBadge } from '../../utils';

export default (Grommet) => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet'))
    .description('This is the top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`
    );

  DocumentedGrommet.propTypes = {
    dir: PropTypes.oneOf(['rtl', 'ltr']).description(
      `Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.`
    ),
    theme: PropTypes.object.description('Custom styles for Grommet app component.'),
  };

  return DocumentedGrommet;
};
