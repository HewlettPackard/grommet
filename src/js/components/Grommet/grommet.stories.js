import React from 'react';
import { storiesOf } from '@storybook/react';

import { Grommet, Anchor, Box } from 'grommet';
import { Add } from 'grommet-icons';

const customTheme = {
  global: {
    colors: {
      custom: '#cc6633',
    },
  },
};


const Themed = () => (
  <Grommet theme={customTheme}>
    <Box pad='medium'>
      <Anchor icon={<Add />} label='Add' color='custom' />
    </Box>
  </Grommet>
);

storiesOf('Grommet', module)
  .add('Theme', () => <Themed />);
