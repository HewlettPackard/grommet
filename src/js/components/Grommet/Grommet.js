import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
