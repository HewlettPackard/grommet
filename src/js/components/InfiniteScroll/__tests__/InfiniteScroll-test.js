import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { InfiniteScroll } from '..';
import { Box } from '../..';

const simpleItems = value =>
  Array(value)
    .fill()
    .map((_, i) => `item ${i + 1}`);

const createPageItems = allChildren => {
  const unfiltered = Array.from(allChildren);
  // Removing any children which are serving as refs
  return unfiltered.filter(childItem => childItem.outerHTML.includes('item'));
};

describe('InfiniteScroll', () => {
  const items = [];
  while (items.length < 4) items.push(items.length);

  test('basic', () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll />
        <InfiniteScroll items={items}>
          {(item, index, ref) => (
            <div ref={ref} key={index}>
              {item}
            </div>
          )}
        </InfiniteScroll>
        <InfiniteScroll items={items}>
          {(item, index) => <div key={index}>{item}</div>}
        </InfiniteScroll>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('step', () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={items} step={2}>
          {(item, index) => <div key={index}>{item}</div>}
        </InfiniteScroll>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('show', () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={items} step={2} show={3}>
          {(item, index) => <div key={index}>{item}</div>}
        </InfiniteScroll>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renderMarker', () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll
          items={items}
          step={2}
          renderMarker={m => <div>{m}</div>}
        >
          {(item, index) => <div key={index}>{item}</div>}
        </InfiniteScroll>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('replace', () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={items} step={2} replace>
          {(item, index) => <div key={index}>{item}</div>}
        </InfiniteScroll>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('Number of Items Rendered', () => {
  test(`Should render items equal to the length of 
  step when step < items.length`, () => {
    const step = 50;
    const { container } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(1000)}
          // show={117}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    const pageItems = createPageItems(container.firstChild.children);
    const expectedItems = step;
    expect(pageItems.length).toEqual(expectedItems);
  });

  test(`Should render items equal to the length of 
  step when step = array.length`, () => {
    const step = 200;
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={simpleItems(200)} step={step}>
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    const pageItems = createPageItems(container.firstChild.children);
    const expectedItems = step;
    expect(pageItems.length).toEqual(expectedItems);
  });

  test(`Should render items equal to the length of 
  item array when step > array`, () => {
    const numItems = 1000;
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={simpleItems(numItems)} step={1050}>
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    const pageItems = createPageItems(container.firstChild.children);
    const expectedItems = numItems;
    expect(pageItems.length).toEqual(expectedItems);
  });

  test(`Should only contain unique items (i.e no duplicates)`, () => {
    const step = 25;
    const numItems = 200;
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={simpleItems(numItems)} step={step}>
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    const pageItems = createPageItems(container.firstChild.children);
    const distinctItems = new Set(pageItems);
    /* Expected number of items should be at the show value rounded
    up to the next step increment/ */
    const expectedItems = step;
    /* If the number of distinct items is equivalent to the length 
    of results, then we have unique items. */
    expect(distinctItems.size).toEqual(expectedItems);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('show scenarios', () => {
  test(`When show, show item should be visible in window`, () => {
    const { container } = render(
      <Grommet>
        <InfiniteScroll items={simpleItems(300)} show={105}>
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );
    // item(104) = 'item 105' because indexing starts at 0.
    // Need to modify this next selection to only be concerned with the
    // visible window.
    const renderedItems = container.firstChild.children.item(104).outerHTML;
    expect(renderedItems).toContain('item 105');
  });

  test(`When show, should only contain unique 
          items (i.e no duplicates)`, () => {
    const step = 25;
    const numItems = 200;
    const showIndex = 67;
    const { container } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(numItems)}
          show={showIndex}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    const pageItems = createPageItems(container.firstChild.children);
    const distinctItems = new Set(pageItems);
    /* Expected number of items should be at the show value rounded
    up to the next step increment/ */
    const expectedItems = Math.ceil(showIndex / step) * step;
    /* If the number of distinct items is equivalent to the length 
    of results, then we have unique items. */
    expect(distinctItems.size).toEqual(expectedItems);
    expect(container.firstChild).toMatchSnapshot();
  });

  test(`should display specified item when show is greater than step`, () => {
    const step = 8;
    const numItems = 200;
    const showIndex = 41;
    const { container, getByText } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(numItems)}
          show={showIndex}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    // Check to see that expected item exists
    const expectedItem = getByText('item 42').innerHTML;
    expect(expectedItem).toMatch(simpleItems(numItems)[showIndex]);

    // Check to see that we have the total number of items we expect
    const pageItems = createPageItems(container.firstChild.children);
    /* Expected number of items should be at the show value rounded
    up to the next step increment/ */
    const expectedItems = Math.ceil(showIndex / step) * step;
    expect(pageItems.length).toEqual(expectedItems);
  });

  test(`should display specified item when show is less than step`, () => {
    const step = 30;
    const numItems = 200;
    const showIndex = 26;
    const { container, getByText } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(numItems)}
          show={showIndex}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    // Check to see that expected item exists
    const expectedItem = getByText('item 27').innerHTML;
    expect(expectedItem).toMatch(simpleItems(numItems)[showIndex]);

    // Check to see that we have the total number of items we expect
    const pageItems = createPageItems(container.firstChild.children);

    /* Expected number of items should be at the show value rounded
    up to the next step increment/ */
    const expectedItems = Math.ceil(showIndex / step) * step;
    expect(pageItems.length).toEqual(expectedItems);
  });

  test(`should display specified item when show is 
        greater than step and replace is true`, () => {
    const step = 18;
    const numItems = 200;
    const showIndex = 88;
    const { container, getByText } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(numItems)}
          replace
          show={showIndex}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    // Check to see that expected item exists
    const expectedItem = getByText('item 89').innerHTML;
    expect(expectedItem).toMatch(simpleItems(numItems)[showIndex]);

    // Check to see that our replace items have been removed from the DOM.
    expect(container.firstChild).not.toContain('item 7');

    /* Check to see that we have the total number of items we expect.
     * When replace is true, the expected number of items should be less
     * than or equal to the step * 2.
     */

    /*
     * The following needs to be uncommented to for this test to pass.
     */
    // const pageItems = createPageItems(container.firstChild.children);
    // const expectedItems = step * 2;
    // expect(pageItems.length).toBeLessThanOrEqual(expectedItems);
  });

  test(`should display specified item when show is 
        less than step and replace is true`, () => {
    const step = 30;
    const numItems = 200;
    const showIndex = 26;
    const { container, getByText } = render(
      <Grommet>
        <InfiniteScroll
          items={simpleItems(numItems)}
          replace
          show={showIndex}
          step={step}
        >
          {item => <Box key={item}>{item}</Box>}
        </InfiniteScroll>
      </Grommet>,
    );

    // Check to see that expected item exists
    const expectedItem = getByText('item 27').innerHTML;
    expect(expectedItem).toMatch(simpleItems(numItems)[showIndex]);

    /* Check to see that we have the total number of items we expect.
     * When replace is true, the expected number of items should be less
     * than or equal to the step * 2.
     */
    const pageItems = createPageItems(container.firstChild.children);
    const expectedItems = step * 2;
    expect(pageItems.length).toBeLessThanOrEqual(expectedItems);
  });
});
