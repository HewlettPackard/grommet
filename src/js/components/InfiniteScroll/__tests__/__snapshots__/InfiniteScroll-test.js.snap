// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;

exports[`InfiniteScroll renderMarker 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    <div
      class="c1"
    />
  </div>
</div>
`;

exports[`InfiniteScroll replace 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div
    class="c1"
  />
</div>
`;

exports[`InfiniteScroll show 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div
    class="c1"
  />
</div>
`;

exports[`InfiniteScroll step 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div
    class="c1"
  />
</div>
`;

exports[`Number of Items Rendered Should only contain unique items (i.e no duplicates) 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    item 1
  </div>
  <div
    class="c1"
  >
    item 2
  </div>
  <div
    class="c1"
  >
    item 3
  </div>
  <div
    class="c1"
  >
    item 4
  </div>
  <div
    class="c1"
  >
    item 5
  </div>
  <div
    class="c1"
  >
    item 6
  </div>
  <div
    class="c1"
  >
    item 7
  </div>
  <div
    class="c1"
  >
    item 8
  </div>
  <div
    class="c1"
  >
    item 9
  </div>
  <div
    class="c1"
  >
    item 10
  </div>
  <div
    class="c1"
  >
    item 11
  </div>
  <div
    class="c1"
  >
    item 12
  </div>
  <div
    class="c1"
  >
    item 13
  </div>
  <div
    class="c1"
  >
    item 14
  </div>
  <div
    class="c1"
  >
    item 15
  </div>
  <div
    class="c1"
  >
    item 16
  </div>
  <div
    class="c1"
  >
    item 17
  </div>
  <div
    class="c1"
  >
    item 18
  </div>
  <div
    class="c1"
  >
    item 19
  </div>
  <div
    class="c1"
  >
    item 20
  </div>
  <div
    class="c1"
  >
    item 21
  </div>
  <div
    class="c1"
  >
    item 22
  </div>
  <div
    class="c1"
  >
    item 23
  </div>
  <div
    class="c1"
  >
    item 24
  </div>
  <div
    class="c1"
  >
    item 25
  </div>
  <div
    class="c2"
  />
</div>
`;

exports[`show scenarios When show, should only contain unique 
          items (i.e no duplicates) 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    item 1
  </div>
  <div
    class="c1"
  >
    item 2
  </div>
  <div
    class="c1"
  >
    item 3
  </div>
  <div
    class="c1"
  >
    item 4
  </div>
  <div
    class="c1"
  >
    item 5
  </div>
  <div
    class="c1"
  >
    item 6
  </div>
  <div
    class="c1"
  >
    item 7
  </div>
  <div
    class="c1"
  >
    item 8
  </div>
  <div
    class="c1"
  >
    item 9
  </div>
  <div
    class="c1"
  >
    item 10
  </div>
  <div
    class="c1"
  >
    item 11
  </div>
  <div
    class="c1"
  >
    item 12
  </div>
  <div
    class="c1"
  >
    item 13
  </div>
  <div
    class="c1"
  >
    item 14
  </div>
  <div
    class="c1"
  >
    item 15
  </div>
  <div
    class="c1"
  >
    item 16
  </div>
  <div
    class="c1"
  >
    item 17
  </div>
  <div
    class="c1"
  >
    item 18
  </div>
  <div
    class="c1"
  >
    item 19
  </div>
  <div
    class="c1"
  >
    item 20
  </div>
  <div
    class="c1"
  >
    item 21
  </div>
  <div
    class="c1"
  >
    item 22
  </div>
  <div
    class="c1"
  >
    item 23
  </div>
  <div
    class="c1"
  >
    item 24
  </div>
  <div
    class="c1"
  >
    item 25
  </div>
  <div
    class="c1"
  >
    item 26
  </div>
  <div
    class="c1"
  >
    item 27
  </div>
  <div
    class="c1"
  >
    item 28
  </div>
  <div
    class="c1"
  >
    item 29
  </div>
  <div
    class="c1"
  >
    item 30
  </div>
  <div
    class="c1"
  >
    item 31
  </div>
  <div
    class="c1"
  >
    item 32
  </div>
  <div
    class="c1"
  >
    item 33
  </div>
  <div
    class="c1"
  >
    item 34
  </div>
  <div
    class="c1"
  >
    item 35
  </div>
  <div
    class="c1"
  >
    item 36
  </div>
  <div
    class="c1"
  >
    item 37
  </div>
  <div
    class="c1"
  >
    item 38
  </div>
  <div
    class="c1"
  >
    item 39
  </div>
  <div
    class="c1"
  >
    item 40
  </div>
  <div
    class="c1"
  >
    item 41
  </div>
  <div
    class="c1"
  >
    item 42
  </div>
  <div
    class="c1"
  >
    item 43
  </div>
  <div
    class="c1"
  >
    item 44
  </div>
  <div
    class="c1"
  >
    item 45
  </div>
  <div
    class="c1"
  >
    item 46
  </div>
  <div
    class="c1"
  >
    item 47
  </div>
  <div
    class="c1"
  >
    item 48
  </div>
  <div
    class="c1"
  >
    item 49
  </div>
  <div
    class="c1"
  >
    item 50
  </div>
  <div
    class="c1"
  >
    item 51
  </div>
  <div
    class="c1"
  >
    item 52
  </div>
  <div
    class="c1"
  >
    item 53
  </div>
  <div
    class="c1"
  >
    item 54
  </div>
  <div
    class="c1"
  >
    item 55
  </div>
  <div
    class="c1"
  >
    item 56
  </div>
  <div
    class="c1"
  >
    item 57
  </div>
  <div
    class="c1"
  >
    item 58
  </div>
  <div
    class="c1"
  >
    item 59
  </div>
  <div
    class="c1"
  >
    item 60
  </div>
  <div
    class="c1"
  >
    item 61
  </div>
  <div
    class="c1"
  >
    item 62
  </div>
  <div
    class="c1"
  >
    item 63
  </div>
  <div
    class="c1"
  >
    item 64
  </div>
  <div
    class="c1"
  >
    item 65
  </div>
  <div
    class="c1"
  >
    item 66
  </div>
  <div
    class="c1"
  >
    item 67
  </div>
  <div
    class="c1"
  >
    item 68
  </div>
  <div
    class="c1"
  >
    item 69
  </div>
  <div
    class="c1"
  >
    item 70
  </div>
  <div
    class="c1"
  >
    item 71
  </div>
  <div
    class="c1"
  >
    item 72
  </div>
  <div
    class="c1"
  >
    item 73
  </div>
  <div
    class="c1"
  >
    item 74
  </div>
  <div
    class="c1"
  >
    item 75
  </div>
  <div
    class="c2"
  />
</div>
`;
